### PR TITLE
Development: Stage 3: Local storage

### PR DESCRIPTION
**Pull request by Vgonma**
# Changes 

- Add local storage functionality

# Bug Fixes

- `Clear all` now clears all tasks correctly

### CODE REVIEW

**Review by Bondok6 — CHANGES_REQUESTED** [review]
## Hi @Vgonma 🙋‍♂️, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No Linter errors ✅
- Your repo is professional ✅
- Implement a function for the "Clear all completed" ✅


## Required Changes ♻️
- [x] Great job using the new README template 💯 However I would suggest updating the project name and project description in your README file 

- [x] I'm getting this error when I run your project, I think it comes from this function `populateList(toDoList);`, kindly make sure that there are no errors in the console, and your app works as expected 👍

image


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js [hunk 1]
```diff
@@ -2,24 +2,30 @@ import './style.css';
...
-toDoList.addTask(new Task('task 3', 1));
-toDoList.addTask(new Task('task 4', 0));
+toDoList.taskList = storage.load('tasks');
+
+// test variables if the list is empty
+// toDoList.addTask(new Task('task 1', 3));
+// toDoList.addTask(new Task('task 2', 2));
+// toDoList.addTask(new Task('task 3', 1));
+// toDoList.addTask(new Task('task 4', 0));
+
```
Comment (TedLivist) [inline src/index.js]: - [x] Kindly remove commented code.

File: src/index.js [hunk 2]
```diff
@@ -28,22 +34,43 @@ form.addEventListener('submit', (e) => {
...
 });
 
 const checkboxes = document.querySelectorAll('.checkbox');
 checkboxes.forEach((box) => {
   box.addEventListener('change', () => {
     const boxIndex = Array.prototype.indexOf.call(checkboxes, box);
-    toDoList.taskList[boxIndex].toggleCompleted();
+    toDoList.taskList[boxIndex].completed = !toDoList.taskList[boxIndex].completed;
   });
 });
```
Comment (TedLivist) [inline src/index.js]: - [x] When I check a checkbox it should change the `completed` value to `true` in the `localStorage` and back to `false` when I uncheck it but it does not happen. Kindly fix that.

File: src/modules/ToDoList.js [hunk 3]
```diff
@@ -34,13 +34,12 @@ export default class ToDoList {
...
-    const arr = [];
-    for (let i = 0; i < this.taskList.length; i += 1) {
-      if (!this.taskList[i].isCompleted()) {
-        arr.push(this.taskList[i]);
+    const len = this.taskList.length;
+    for (let i = len - 1; i >= 0; i -= 1) {
+      if (this.taskList[i].completed) {
+        this.removeTask(i);
       }
     }
```
Comment (TedLivist) [inline src/modules/ToDoList.js]: - [x] Kindly refactor your clear completed functionality to use the `filter()` method.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Vgonma,

✔️ Tasks can be added and removed

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AmaduKamara — APPROVED** [review]
# Status: Approved 

Hi @Vgonma,

Good job making the changes highlighted by the previous reviewers 👍 

Your project is complete!  

## Highlights:
✅ You used Gitflow correctly 👍 
✅ Linter checks passed 👍 
✅ The todo completed state changes as user interacts with the completed and uncompleted state 👍 

There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**